### ISSUE TITLE
object_usage_linter doesn't catch assignments with =

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1081]
Noticed while working on #1078.

Compare

```
  expect_lint(
    trim_some("
      fun <- function() {
        a <- 1
      }
    "),
    rex("local variable", anything, "assigned but may not be used"),
    linter
  )
```

and

```
  expect_lint(
    trim_some("
      fun = function() {
        a = 1
      }
    "),
    rex("local variable", anything, "assigned but may not be used"),
    linter
  )
```

The latter test fails on HEAD because the xpath for `=` assignments is wrong. Fix forthcoming.
